### PR TITLE
Add progress indication, fixes #394

### LINKED ISSUE #394: Add progress indication
- borg check (repository part) #195
- segment replay #188
- upgrade (refactor existing progress indication code)
- (delete #377 -- removed this, no easy way)
- others, anything that could take long if a slow machine or connection is involved

See also #123, #124, #264 and #291.

This ticket supercedes #195, #188, #377.

***

:moneybag: [there is a bounty for this](https://www.bountysource.com/issues/28080559-add-progress-indication)

### PR DESCRIPTION
**Pull request by ThomasWaldmann**

### CODE REVIEW

**Comment by RonnyPfannschmidt** [comment]
It's a bit sad that practically everyone reimplements that in some way

File: borg/helpers.py [hunk 1]
```diff
@@ -890,6 +890,82 @@ def yes(msg=None, retry_msg=None, false_msg=None, true_msg=None,
...
 
 
+class ProgressIndicatorPercent:
+    def __init__(self, total, step=5, start=0, same_line=False, msg="%3.0f%%", file=sys.stderr):
+        """
+        Percentage-based progress indicator
+
+        :param total: total amount of items
+        :param step: step size in percent
+        :param start: at which percent value to start
```
Comment (skorokithakis) [inline borg/helpers.py]: Some parameters are missing here.

File: borg/helpers.py [hunk 2]
```diff
@@ -890,6 +890,82 @@ def yes(msg=None, retry_msg=None, false_msg=None, true_msg=None,
...
+        self.total = total
+        self.trigger_at = start  # output next percentage value when reaching (at least) this
+        self.step = step
+        self.file = file
+        self.msg = msg
+        self.same_line = same_line
+
+    def progress(self, current=None):
+        if current is None:
+            current = self.counter
```
Comment (skorokithakis) [inline borg/helpers.py]: Why not just use self.counter here regardless?

Comment (ThomasWaldmann) [inline borg/helpers.py]: well, if you'ld like to restart from a specific progress (not from 0), you can just give it via "current".
but you don't need to, if you just start from 0 and proceed step by step.

Comment (skorokithakis) [inline borg/helpers.py]: I meant this:

```
if current is not None:
    self.counter = current
pct = self.counter * 100 / self.total
self.counter += 1
```

Comment (ThomasWaldmann) [inline borg/helpers.py]: ok

File: borg/helpers.py (hunk 1 above)
Comment (ThomasWaldmann) [inline borg/helpers.py]: ok.

**Pull request merged**